### DISCUSSION TITLE
Combo counter should be directly above the catcher in non-legacy skins (i.e. "argon" and "triangles")

### DISCUSSION THREAD

**Discussion by Xinnoh** [ppy/osu#21818] [Ideas]
When playing at higher approach rates (greater than 9), all players keep their view on the top half of the screen.
When getting the position of the catcher, we don't look at the catcher, we look at the combo.
The combo count (or combo fire if you hide ui) lets us know where it is without looking at the bottom of the screen.

The skin customisation doesn't let you attach objects to the catcher so that's not an option.

As it stands it's borderline unplayable on higher AR specifically because of this. We need some way to know the catcher's position while keeping attention focused on the top half of the screen.
(the other critical issue being that the catcher is so vertically tiny)

![screenshot007](https://user-images.githubusercontent.com/30382015/209708836-a5fdcc85-02ef-4675-a8cc-d3507155f266.jpg)

![osu_2022-12-27_10-39-06](https://user-images.githubusercontent.com/30382015/209708827-a6644325-c075-414c-a3c5-c61a736c68c5.jpg)

**Comment by bdach** [comment]
The combo count and hit explosions are already above the catcher in the classic skin.

**Reply by Xinnoh** [reply]
then would make sense to implement an option like this in argon

**Reply by peppy** [reply]
Maybe, we'll see how feedback is going forward. There are still overlay UI elements to implement for the argon skin which haven't hit yet.

**Reply by yamplum** [reply]
Just throwing an idea out there, but could it be possible to add the catcher as an anchor point for UI elements? (Perhaps on a different layer to static overlay.) Could have interesting applications in other modes as well, like whacking a combo counter on the cursor in std.

**Reply by bdach** [reply]
@yamplum definitely, but there is a fair bit of work required before we can get there (ruleset-specific skin layouts is the first thing that needs to happen).
